util/conditions/v1beta2/summary.go

```diff
@@ -0,0 +1,134 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta2
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// SummaryOption is some configuration that modifies options for a summary call.
+type SummaryOption interface {
+	// ApplyToSummary applies this configuration to the given summary options.
+	ApplyToSummary(*SummaryOptions)
+}
+
+// SummaryOptions allows to set options for the summary operation.
+type SummaryOptions struct {
+	mergeStrategy                  MergeStrategy
+	conditionTypes                 []string
+	negativePolarityConditionTypes []string
+	ignoreTypesIfMissing           []string
+	stepCounter                    bool
+}
+
+// ApplyOptions applies the given list options on these options,
+// and then returns itself (for convenient chaining).
+func (o *SummaryOptions) ApplyOptions(opts []SummaryOption) *SummaryOptions {
+	for _, opt := range opts {
+		opt.ApplyToSummary(o)
+	}
+	return o
+}
+
+// NewSummaryCondition creates a new condition by summarizing a set of conditions from an object.
+// If any of the condition in scope does not exist in the source object, missing conditions are considered Unknown, reason NotYetReported.
+//
```

Comment:
Let's also mention at least the ignoreTypesIfMissing option. Ideally we should mention all valid options (same for the other funcs).

I think that makes it easier to discover which options are available per func and what they do in each util func

Reply:
Done